core/citrus-base/src/main/java/org/citrusframework/validation/matcher/core/UuidV4ValidationMatcher.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.citrusframework.validation.matcher.core;
+
+import org.citrusframework.context.TestContext;
+import org.citrusframework.exceptions.ValidationException;
+import org.citrusframework.validation.matcher.ValidationMatcher;
+
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * UuidValidationMatcher checks if valid UUID version 4 is present
+ */
+public class UuidV4ValidationMatcher implements ValidationMatcher {
+    @Override
+    public void validate(String fieldName, String value, List<String> controlParameters, TestContext context) throws ValidationException {
+        try {
+            UUID uuid = UUID.fromString(value);
+
+            if (uuid.version() != 4) {
+                extracted(fieldName, value);
+            }
+        } catch (IllegalArgumentException e) {
+            extracted(fieldName, value);
+        }
+    }
+
+    private void extracted(String fieldName, String value) {
```

Comment:
this name is auto-generated, please adapt it.

```suggestion
    private void throwValidationException(String fieldName, String value) {
```

Reply:
changed method name